Assets/Scripts/UI Scripts/WinLoseHandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class WinLoseHandler : MonoBehaviour
+{
+    //Eventually these will probably be moved to a singleton
+    [SerializeField] GameObject goal;
+    [SerializeField] GameObject hazards;
+    private int alive;
+    private int dead;
+
+
+    public static event Action<string> WinLoseTriggered;
+
+    void Update()
+    {
+        alive = goal.GetComponent<EndGoal>().FrogGoal;
```

Comment:
Don't use GetComponent in Update, it's really expensive. Do it in Start or Awake and cache the result (aka store it as a variable so you don't need to do it again)